polkadot/node/core/candidate-validation/src/lib.rs

```diff
@@ -169,7 +189,37 @@ where
 			..
 		} => async move {
 			let _timer = metrics.time_validate_from_exhaustive();
+			let relay_parent = candidate_receipt.descriptor.relay_parent();
+
+			let Some(claim_queue) = claim_queue(relay_parent, &mut sender).await else {
+				let error = "cannot fetch the claim queue from the runtime";
+
+				gum::warn!(
+					target: LOG_TARGET,
+					?relay_parent,
+					error
+				);
+
+				let _ = response_sender.send(Err(ValidationFailed(error.into())));
+				return
+			};
+
+			let Ok(Ok(expected_session_index)) =
+				util::request_session_index_for_child(relay_parent, &mut sender).await.await
```

Comment:
what about validations happenning for remote disputes? for forks we haven't imported this should fail.
wasn't the idea with the session index in the candidate descriptor that we won't check it for remote disputes?

also true for the claim queue. we shouldn't return errors in this case

Reply:
Yeah, we only need all of these checks for backing only, otherwise no need to, because the candidate has been already included in some fork.